evm/utils/test_builder/test_builder.py

```diff
@@ -0,0 +1,371 @@
+from collections import (
+    defaultdict,
+)
+from functools import (
+    partial,
+)
+
+from evm.db.state import (
+    MainAccountStateDB,
+    ShardingAccountStateDB,
+)
+from evm.utils.fixture_tests import (
+    hash_log_entries,
+)
+
+from cytoolz import (
+    assoc,
+    assoc_in,
+    curry,
+    merge,
+)
+from eth_utils import (
+    apply_formatters_to_dict,
+    decode_hex,
+    encode_hex,
+    to_canonical_address,
+)
+
+from .normalization import (
+    normalize_bytes,
+    normalize_call_creates,
+    normalize_environment,
+    normalize_execution,
+    normalize_int,
+    normalize_logs,
+    normalize_state,
+    normalize_transaction,
+    normalize_transaction_group,
+    normalize_networks,
+)
+from .builder_utils import (
+    add_transaction_to_group,
+    calc_state_root,
+    compile_vyper_lll,
+    get_test_name,
+    get_version_from_git,
+    deep_merge,
+    wrap_in_list,
+)
+from .formatters import (
+    filled_state_test_formatter,
+    filled_vm_test_formatter,
+)
+
+
+#
+# Defaults
+#
+
+DEFAULT_MAIN_ENVIRONMENT = {
+    "currentCoinbase": to_canonical_address("0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba"),
+    "currentDifficulty": 131072,
+    "currentGasLimit": 1000000,
+    "currentNumber": 1,
+    "currentTimestamp": 1000,
+    "previousHash": decode_hex(
+        "0x5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6"
+    ),
+}
+
+DEFAULT_SHARDING_ENVIRONMENT = {
+    "shardID": 0,
+    "expectedPeriodNumber": 0,
+    "periodStartHash": decode_hex(
+        "0x148067ef259ce711201e6b2a8438b907d0ac0549deef577aff58f1b9143a134a"
+    ),
+    "currentCoinbase": to_canonical_address("0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba"),
+    "currentNumber": 1,
+    "previousHash": decode_hex(
+        "0x5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6"
+    ),
+}
+
+
+DEFAULT_MAIN_TRANSACTION = {
+    "data": b"",
+    "gasLimit": 100000,
+    "gasPrice": 0,
+    "nonce": 0,
+    "secretKey": decode_hex("0x45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8"),
+    "to": to_canonical_address("0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6"),
```

Comment:
Did this address come from anywhere in particular?

Reply:
It's used in a lot of existing JSON tests (https://github.com/ethereum/tests/search?utf8=%E2%9C%93&q=0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6&type=), but I guess they just generated it randomly.